types/types.go

```diff
@@ -218,6 +225,23 @@ type BlobInfoCache interface {
 	CandidateLocations(transport ImageTransport, scope BICTransportScope, digest digest.Digest, canSubstitute bool) []BICReplacementCandidate
 }
 
+// BlobInfoCache2 extends BlobInfoCache by adding the ability to track information about what kind of compression was applied to
+// the blobs it keeps information about.  If a BlobInfoCache also implements BlobInfoCache2, users of the cache are encouraged to
+// use RecordKnownLocation2() instead of RecordKnownLocation(), and to use CandidateLocations2() instead of CandidateLocations().
+type BlobInfoCache2 interface {
+	BlobInfoCache
+	// RecordKnownLocation2 records that a blob with the specified digest exists within the specified (transport, scope) scope,
+	// compressed with the specified compression algorithm, and can be reused given the opaque location data.
```

Comment:
If this can represent “uncompressed” and “unknown”, the two should be documented and distinct.

(If we end up with an internal definition of the interface, perhaps we can avoid the “unknown” case.)

Reply:
Added distinct values for both Uncompressed and Unknown; for Unknown we just drop the entry from the blobinfocache.